Content.Client/Cargo/UI/BountyHistoryEntry.xaml.cs

```diff
@@ -0,0 +1,54 @@
+using Content.Client.Message;
+using Content.Shared.Cargo;
+using Content.Shared.Cargo.Prototypes;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Timing;
+
+namespace Content.Client.Cargo.UI;
+
+[GenerateTypedNameReferences]
+public sealed partial class BountyHistoryEntry : BoxContainer
+{
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+
+    public BountyHistoryEntry(CargoBountyHistoryData bounty)
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        if (!_prototype.TryIndex<CargoBountyPrototype>(bounty.Bounty, out var bountyPrototype))
+            return;
+
+        var items = new List<string>();
+        foreach (var entry in bountyPrototype.Entries)
+        {
+            items.Add(Loc.GetString("bounty-console-manifest-entry",
+                ("amount", entry.Amount),
+                ("item", Loc.GetString(entry.Name))));
+        }
```

Comment:
Totally optional, but you could use LINQ for this.

```suggestion
        var items = bountyPrototype.Entries
            .Select(entry => Loc.GetString("bounty-console-manifest-entry",
                ("amount", entry.Amount),
                ("item", Loc.GetString(entry.Name))));
```

Reply:
Tried this but I'll come back to it, I plan on looking at adding a history to the main request board as well, if possible. So given what you've mentioned before about "legacy" cargo code, I was thinking of cleaning that up before going any further with it. This'll probably fall into that :)